lib/auth/auth_with_roles_test.go

```diff
@@ -3645,3 +3645,139 @@ func TestListResources_WithRoles(t *testing.T) {
 		})
 	}
 }
+
+// TestGenerateHostCert attempts to generate host certificates using various
+// RBAC rules
+func TestGenerateHostCert(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	srv := newTestTLSServer(t)
+
+	clusterName := srv.ClusterName()
+
+	_, pub, err := native.GenerateKeyPair()
+	require.NoError(t, err)
+
+	for _, test := range []struct {
+		desc       string
+		principals []string
+		skipRule   bool
+		where      string
+		deny       bool
+		denyWhere  string
+		expect     func(error) bool
```

Comment:
nit: I'm not sure if anyone else gets this, but I do find the test table quite hard to read when the expect is omitted for success cases. Not a major issue, just something that came to mind.

Reply:
I'll add a quick `noError` helper function.

Reply:
Perfect thanks.